### PR TITLE
Created ERC20 token, add some tests, add package.json

### PR DESCRIPTION
**Pull request by Norfolks**

### CODE REVIEW

File: test/test_opet_coin.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+            assert.ifError('Error, too much tokens successfuly airdropped');
+        } catch (err) {
+            assert.equal(err, 'Error: VM Exception while processing transaction: revert', "Wrong error too big amount of airdrop");
+        }
+    });
+});
+
+
+contract("Opet Token", async function (accounts) {
+    it("test transfer unable to execute", async function () {
```
Comment (serebrov) [inline test/test_opet_coin.js]: better `test transfer cannot be used` or `test transfer paused`

File: test/test_opet_coin.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        let token = await Token.deployed();
+        let mintDestination = accounts[1];
+        let transferDestination = accounts[2];
+
+
+        await token.mint(mintDestination, 2);
+        try {
+            await token.transfer(transferDestination, 1, {'from': mintDestination});
+            assert.ifError('Error, transfer should be unable');
```
Comment (serebrov) [inline test/test_opet_coin.js]: `Error, transfer should not be possible`

File: test/test_opet_coin.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+
+contract("Opet Token", async function (accounts) {
+    it("test transfer unable to execute", async function () {
+        let token = await Token.deployed();
+        let mintDestination = accounts[1];
+        let transferDestination = accounts[2];
+
+
+        await token.mint(mintDestination, 2);
```
Comment (serebrov) [inline test/test_opet_coin.js]: Please add a comment above the block - `Test token "transfer" flow`

File: test/test_opet_coin.js [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        try {
+            await token.transfer(transferDestination, 1, {'from': mintDestination});
+            assert.ifError('Error, transfer should be unable');
+        } catch (err) {
+            assert.equal(err, 'Error: VM Exception while processing transaction: revert', "Transfer works before token unpaused");
+        }
+
+        await token.approve(transferDestination, 1, {'from': mintDestination});
```
Comment (serebrov) [inline test/test_opet_coin.js]: Please add a comment above the block - `Test token "approve" / "transferFrom" flow`

File: contracts/Opet_coin.sol [hunk 5]
```diff
@@ -0,0 +1,329 @@
...
+    emit Mint(_to, _amount);
+    emit Transfer(address(0), _to, _amount);
+    return true;
+  }
+
+  /**
+   * @dev Function to stop minting new tokens.
+   * @return True if the operation was successful.
+   */
+  function finishMinting() onlyOwner canMint public returns (bool) {
```
Comment (serebrov) [inline contracts/Opet_coin.sol]: finishMinting is not covered by tests

File: contracts/Opet_coin.sol [hunk 6]
```diff
@@ -0,0 +1,329 @@
...
+  using SafeMath for uint256;
+
+  mapping(address => uint256) balances;
+
+  uint256 totalSupply_;
+
+  /**
+  * @dev total number of tokens in existence
+  */
+  function totalSupply() public view returns (uint256) {
```
Comment (serebrov) [inline contracts/Opet_coin.sol]: totalSupply() is not covered by tests

File: contracts/Opet_coin.sol [hunk 7]
```diff
@@ -0,0 +1,329 @@
...
+   * @dev Increase the amount of tokens that an owner allowed to a spender.
+   *
+   * approve should be called when allowed[_spender] == 0. To increment
+   * allowed value is better to use this function to avoid 2 calls (and wait until
+   * the first transaction is mined)
+   * From MonolithDAO Token.sol
+   * @param _spender The address which will spend the funds.
+   * @param _addedValue The amount of tokens to increase the allowance by.
+   */
+  function increaseApproval(
```
Comment (serebrov) [inline contracts/Opet_coin.sol]: increaseApproval is not covered by tests

File: contracts/Opet_coin.sol [hunk 8]
```diff
@@ -0,0 +1,329 @@
...
+   * @dev Decrease the amount of tokens that an owner allowed to a spender.
+   *
+   * approve should be called when allowed[_spender] == 0. To decrement
+   * allowed value is better to use this function to avoid 2 calls (and wait until
+   * the first transaction is mined)
+   * From MonolithDAO Token.sol
+   * @param _spender The address which will spend the funds.
+   * @param _subtractedValue The amount of tokens to decrease the allowance by.
+   */
+  function decreaseApproval(
```
Comment (serebrov) [inline contracts/Opet_coin.sol]: decreaseApproval is not covered by tests

File: contracts/Ownable.sol [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+  modifier onlyOwner() {
+    require(msg.sender == owner);
+    _;
+  }
+
+  /**
+   * @dev Allows the current owner to transfer control of the contract to a newOwner.
+   * @param newOwner The address to transfer ownership to.
+   */
+  function transferOwnership(address newOwner) public onlyOwner {
```
Comment (serebrov) [inline contracts/Ownable.sol]: transferOwnership() is not covered by tests

File: contracts/Ownable.sol [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+    require(newOwner != address(0));
+    emit OwnershipTransferred(owner, newOwner);
+    owner = newOwner;
+  }
+
+  /**
+   * @dev Allows the current owner to relinquish control of the contract.
+   */
+  function renounceOwnership() public onlyOwner {
```
Comment (serebrov) [inline contracts/Ownable.sol]: renounceOwnership() is not covered by tests

File: contracts/Opet_coin.sol [hunk 11]
```diff
@@ -0,0 +1,329 @@
...
+ * @dev Simple ERC20 Token example, with mintable token creation
+ * @dev Issue: * https://github.com/OpenZeppelin/openzeppelin-solidity/issues/120
+ * Based on code by TokenMarketNet: https://github.com/TokenMarketNet/ico/blob/master/contracts/MintableToken.sol
+ */
+contract MintableToken is StandardToken, Ownable {
+  event Mint(address indexed to, uint256 amount);
+  event MintFinished();
+
+  bool public mintingFinished = false;
+  uint256 public maxMintLimit;
```
[the remark this answers is not in the document]
Comment (Norfolks) [reply contracts/Opet_coin.sol]: yes

File: contracts/Opet_coin.sol [hunk 12]
```diff
@@ -0,0 +1,329 @@
...
+ * @title Mintable token
+ * @dev Simple ERC20 Token example, with mintable token creation
+ * @dev Issue: * https://github.com/OpenZeppelin/openzeppelin-solidity/issues/120
+ * Based on code by TokenMarketNet: https://github.com/TokenMarketNet/ico/blob/master/contracts/MintableToken.sol
+ */
+contract MintableToken is StandardToken, Ownable {
+  event Mint(address indexed to, uint256 amount);
+  event MintFinished();
+
+  bool public mintingFinished = false;
```
[the remark this answers is not in the document]
Comment (Norfolks) [reply contracts/Opet_coin.sol]: There is comment at finishMinting function

**Pull request merged**